src/main/java/org/zalando/nakadi/service/EventStreamWriterBinary.java

```diff
@@ -48,7 +68,7 @@ public int writeBatch(final OutputStream os, final Cursor cursor, final List<Str
         if (!events.isEmpty()) {
             os.write(B_EVENTS_ARRAY_BEGIN);
             for (int i = 0; i < events.size(); i++) {
-                final byte[] event = events.get(i).getBytes(StandardCharsets.UTF_8);
+                final byte[] event = extractor.apply(events.get(i)).getBytes(StandardCharsets.UTF_8);
```

Comment:
@rcillo what do you think about having slightly more code duplication in favour avoiding type casting ?